src/topology/builder.rs

```diff
@@ -512,6 +512,16 @@ impl<'a> Builder<'a> {
             let typetag = sink.inner.get_component_name();
             let input_type = sink.inner.input().data_type();
 
+            let span = error_span!(
+                "sink",
+                component_kind = "sink",
+                component_id = %key.id(),
+                component_type = %sink.inner.get_component_name(),
+                // maintained for compatibility
+                component_name = %key.id(),
+            );
+            let _entered_span = span.enter();
```

Comment:
We might also want to remove the creation of `buffer_span` below and replace its usage with `Span::current()`, so that we don't have a duplicated span.

Reply:
Good catch. I dropped the duplicate tags from the buffer span in [1bb3954](https://github.com/vectordotdev/vector/pull/17765/commits/1bb39540ddf8bc030002d201a145159cac345479)